ci-e2e-v1.13: Fix workflow

brb opened the pull request:
Unfortunately, 93baa34 was run against the latest Cilium, so it didn't catch v1.13 specific issues (limitations). This commit fixes:

* Use --tunnel instead of --tunnelProtocol. The latter was only introduced in v1.14 \[1\].
* Disable IPv6 for IPsec configuration.
* Downgrade CLI to v0.13.2 until \[2\] has been resolved.
* Remove external CIDRs, as they are not needed for v0.13.2 CLI.
* Disable L7 if EGW or WG is enabled (v1.13 limitation).
* Remove host-to-host encryption configurations (v1.13 doesn't support it).

\[1\]: https://github.com/cilium/cilium/pull/24561
\[2\]: https://github.com/cilium/cilium-cli/issues/1627

Successful run https://github.com/cilium/cilium/actions/runs/4959820195/jobs/8874465740.

brb committed:
ci-e2e-v1.13: Fix the workflow

Unfortunately, 93baa34 was ran against the latest Cilium, so it didn't
catch v1.13 specific issues (limitations). This commit fixes:

* Use --tunnel instead of --tunnelProtocol. The latter was only
  introduced in v1.14 [1].
* Disable IPv6 for IPsec configuration.
* Downgrade CLI to v0.13.2 until [2] has been resolved.
* Remove external CIDRs, as they are not needed for v0.13.2 CLI.
* Disable L7 if EGW or WG is enabled (v1.13 limitation).
* Remove host-to-host encryption configurations (v1.13 doesn't support
  it).

[1]: https://github.com/cilium/cilium/pull/24561
[2]: https://github.com/cilium/cilium-cli/issues/1627

Fixes: 93baa34 ("ci-e2e: backport changes in conformance-e2e into v1.13 tests")
Signed-off-by: Martynas Pumputis <m@lambda.lt>

tklauser commented on .github/workflows/conformance-e2e-v1.13.yaml:
```diff
@@ -65,7 +65,7 @@ concurrency:
 
 env:
   # renovate: datasource=github-releases depName=cilium/cilium-cli
```
We'll need to temporarily remove or modify this comment such that it no longer matches https://github.com/cilium/cilium/blob/3a3f135a7e0d8e4b665362c9f1c8b8e227858d02/.github/renovate.json5#L342 otherwise renovate will bump the version again when it runs next Friday.

thorn3r reviewed (approved):
i can add a commit to disable renovate if @brb is offline for the day now. this PR is blocking [backports](https://github.com/cilium/cilium/pull/25346) that include release blockers

thorn3r committed:
ci-e2e-v1.13: disable renovate

The CLI version has been downgraded to v0.13.2 until
https://github.com/cilium/cilium-cli/issues/1627 has been resolved.

This commit disables renovate on the ci-e2e-v1.13 workflow until the
issue has been resolved to prevent unintentional upgrade.

Signed-off-by: Tim Horner <timothy.horner@isovalent.com>
